### PR TITLE
feat(authentication, types, utils): Add Authentication provider scopes

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Add providers as options for authentication module
- validate provider scope is defined when authenticating
- Type `authenticate` and `verifyCallback`

### CODE REVIEW

File: packages/authentication/integration-tests/__tests__/services/providers/username-password.spec.ts [hunk 1]
```diff
@@ -91,6 +92,7 @@ describe("AuthenticationModuleService - AuthProvider", () => {
 
       const res = await service.authenticate("usernamePassword", {
```
Comment (srindom) [inline packages/authentication/integration-tests/__tests__/services/providers/username-password.spec.ts:93]: would be nice to rename this to something like `userpass` or `user-pass`

Comment (olivermrbl) [reply packages/authentication/integration-tests/__tests__/services/providers/username-password.spec.ts:93]: Or use `local`. Think people will find that familiar from passportjs

Comment (pKorsholm) [reply packages/authentication/integration-tests/__tests__/services/providers/username-password.spec.ts:93]: rename in #6251

File: packages/types/src/authentication/common/provider.ts [hunk 2]
```diff
@@ -2,4 +2,21 @@ export type AuthenticationResponse = {
...
   error?: string
+  location?: string
+}
+
+export type AuthModuleProviderConfig = {
+  name: string
+  scopes: Record<string, AuthProviderScope>
+}
+
+export type AuthProviderScope = { domain?: string } & Record<string, string>
```
Comment (olivermrbl) [inline packages/types/src/authentication/common/provider.ts]: nit(non-blocking): Should this type be entirely open-ended? Are we using `domain` yet?
```suggestion
export type AuthProviderScope = Record<string, unknown>
```

**Review by olivermrbl — APPROVED** [review]
LGTM

**Pull request merged**